tests/test_analysis.py

```diff
@@ -266,6 +267,42 @@ def test_get_mean_fr_per_condition(self):
         np.testing.assert_equal(pvalue[1]<0.05, True)
         np.testing.assert_equal(pvalue[2]>0.05, True)
 
+    def test_calc_dprime(self):
+
+        # Single channel test
+        np.random.seed(0)
+        noise_dist = np.random.normal(0, 1, 1000)
+        signal_dist = np.random.normal(1, 1, 1000)
+        
+        dprime = aopy.analysis.calc_dprime(noise_dist, signal_dist)
+
+        # Recalculate by hand
+        m1 = np.mean(noise_dist, axis=0)
+        m2 = np.mean(signal_dist, axis=0)
+        s1 = np.std(noise_dist, axis=0)
+        s2 = np.std(signal_dist, axis=0)
+        mean_diff = m2 - m1
+        sd_sum = ((len(noise_dist)-1) * s1 + ((len(signal_dist)-1) * s2))
+        sd_pooled = sd_sum/(len(noise_dist) + len(signal_dist) - 2)
+        dprime_hand = mean_diff / sd_pooled
+
+        np.testing.assert_allclose(dprime, dprime_hand)
+
+        # Simple multi-channel test
+        noise_dist = np.array([[0, 1], [0, 1]]).T
+        signal_dist = np.array([[1, 2], [1, 2]]).T
+        print(noise_dist.shape)
+        dprime = aopy.analysis.calc_dprime(noise_dist, signal_dist)
+
+        np.testing.assert_allclose(dprime, np.array([1, 1]))
+
+        # Multi-class test
+        noise_dist = np.array([[0, 1, 0, 1], [0, 1, 0, 1]]).T
+        signal_dist = np.array([[1, 2, 1, 2], [2, 3, 2, 3]]).T
+        dprime = aopy.analysis.calc_dprime(noise_dist, signal_dist)
```

Comment:
Its unclear to me what this is testing that is different from the one above (simple multi-channel test)

Reply:
yeah my mistake, i meant to add another distribution but i just added more trials within each distribution